sentence_transformers/configuration.py

```diff
@@ -0,0 +1,60 @@
+import logging
+from typing import List, Dict, Union
+import transformers
+from sentence_transformers import __version__
+import torch
+import os
+from packaging.version import Version
+
+from transformers import PretrainedConfig
+
+
+logger = logging.getLogger(__name__)
+
+
+class SentenceTransformerConfig(PretrainedConfig):
+    model_type = "sentence-transformers"
+    is_composition = True
+
+    def __init__(self, **kwargs) -> None:
+        self.modules: List[str] = kwargs.pop("modules", [])
+        self.__version__: Dict[str, str] = kwargs.pop(
+            "__version__",
+            {
+                "sentence_transformers": __version__,
+                "transformers": transformers.__version__,
+                "pytorch": torch.__version__,
+            },
+        )
+        super().__init__(**kwargs)
+        self.empty_on_init = len(self.modules) == 0
+
+        if Version(self.__version__["sentence_transformers"]) > Version(__version__):
+            logger.warning(
+                f"You try to use a model that was created with version {self.__version__['sentence_transformers']}, "
+                f"however, your version is {__version__}. This might cause unexpected behavior or errors. "
+                "In that case, try to update to the latest version."
+            )
+
+    def save_pretrained(self, save_directory: Union[str, os.PathLike], push_to_hub: bool = False, **kwargs):
+        """
+        Save a configuration object to the directory `save_directory`, so that it can be re-loaded using the
+        [`~SentenceTransformerConfig.from_pretrained`] class method.
+
+        Args:
+            save_directory (`str` or `os.PathLike`):
+                Directory where the configuration JSON file will be saved (will be created if it does not exist).
+            push_to_hub (`bool`, *optional*, defaults to `False`):
+                Whether or not to push your model to the Hugging Face model hub after saving it. You can specify the
+                repository you want to push to with `repo_id` (will default to the name of `save_directory` in your
+                namespace).
+            kwargs (`Dict[str, Any]`, *optional*):
+                Additional key word arguments passed along to the [`~utils.PushToHubMixin.push_to_hub`] method.
+        """
```

Comment:
It seems like there are two documentation styles in this PR :grin: 

I see the above that looks very close to `transformers` style, but the `save` method in the `SentenceTransformer.py` module has a very different style:

```
        """
        Saves all elements for this seq. sentence embedder into different sub-folders
        :param path: Path on disc
        :param model_name: Optional model name
        :param create_model_card: If True, create a README.md with basic information about this model
        :param train_datasets: Optional list with the names of the datasets used to to train the model
        """
```

I understand this may be legacy, but it'll probably be simpler to choose one and stick with one